phalcon/di/service.zep

```diff
@@ -174,7 +174,13 @@ class Service implements ServiceInterface
 					 * Bounds the closure to the current DI
 					 */
 					if typeof dependencyInjector == "object" {
-						let definition = \Closure::bind(definition, dependencyInjector);
+						let reflectionFunction = new \ReflectionFunction(definition);
+    						if (
+							reflectionFunction->getClosureScopeClass() === null ||
+							reflectionFunction->getClosureThis() !== null
+    						) {
+							let definition = \Closure::bind(definition, dependencyInjector);
+						}
```

Comment:
That's why there should be some test for this, in which cases we want $this in the closure, in which don't.